docs/architecture.md

```diff
@@ -33,4 +33,25 @@ An analyzer is a gRPC service that will be called by the [Server](#server) to pe
 
 They are not part of **source{d} Lookout** repository so they can be developed by third parties.
 
-source{d} Lookout Server will call all the registered Analyzers to produce comments for the opened Pull Request in the watched repositories. To register new Analyzers in the configuration file, `lookoutd` will need to be restarted.
+**source{d} Lookout** Server will call all the registered Analyzers to produce comments for the opened Pull Request in the watched repositories. To register new Analyzers in the configuration file, `lookoutd` will need to be restarted.
+
+
+# External services
+
+**source{d} Lookout** will call some external services, some of them depending on **source{d} Lookout** configuration or on how it is run.
+
+## Babelfish
+
+Babelfish is used to parse files into [UAST](https://doc.bblf.sh/uast/uast-specification.html) when the [DataService](#dataservice) `GetChanges` or `GetFiles` methods are asked to include UASTs.
+
+Babelfish can also be directly called through **DataService** proxy.
+
+## RabbitMQ
+
+If **source{d} Lookout** is run in [distributed mode](github.com/src-d/lookout/docs/how-to-run.md#distributed-mode), the watcher will enqueue in RabbitMQ all new repository and PR events, and the workers will take from there the events to be processed.
+
+## PostgreSQL
+
+**source{d} Lookout** uses PostgreSQL to store application internal state. There will be stored processed events (`ReviewEvent` and `PushEvent`), their targets (`ReviewTarget`), and the comments (`Comment`) returned by the Analyzers.
```

Comment:
My preference is that we should not go into details of what we store and what we use it for. The way I see it if we document it here, in architecture, we are implying you may be interested in the DB contents, or that you may want to manipulate the DB.

I propose
```suggestion
**source{d} Lookout** uses PostgreSQL to store the application internal state.
```

And also to remove below
`It is used to avoid processing the same PR twice for the same push and having duplicate comments.`

Reply:
I think this section is for developers and people interested in the internals and how the app works. Based on that, I find it useful to explain the purpose of the Database and what's inside.
Are you strongly against adding this info? If so, I'll happily drop it :dancing_men:

Reply:
The way I see this section, it's meant to be a high level overview of the architecture, useful for example for the admin that has to deploy it, or the developer of a new analyzer.

Going into detail of what the DB stores can be distracting for someone without the proper context (the knowledge of the internals). If we redefine this section to be meant for developers of this repo, we would probably have to explain how we use kallax for persistency...

Reply:
agree on keeping it high level

Reply:
I believe that saying that it's stored there the comments returned by the analyzers is not low level, and is way high level than explaining how kallax works.
On the other hand, explaining which tables and contents are stored there (even which indexes does it contain) is also useful for admin because it can be useful for them to dimension the service.

Anyhow, I'll drop the parts that you don't like to avoid blocking this PR, and let that part for the future (if I find time or energies).